src/components/Sidebar/Participants/ParticipantsTab.vue

```diff
@@ -0,0 +1,214 @@
+</template>
+
+<script>
+import Caption from '../../Caption'
+import CurrentParticipants from './CurrentParticipants/CurrentParticipants'
+import Hint from '../../Hint'
+import ParticipantsList from './ParticipantsList/ParticipantsList'
+import SearchBox from '../../SearchBox'
+import debounce from 'debounce'
+import { EventBus } from '../../../services/EventBus'
+import { CONVERSATION, WEBINAR } from '../../../constants'
+import { searchPossibleConversations } from '../../../services/conversationsService'
+import { fetchParticipants } from '../../../services/participantsService'
+
+export default {
+	name: 'ParticipantsTab',
+	components: {
+		CurrentParticipants,
+		SearchBox,
+		Caption,
+		Hint,
+		ParticipantsList,
+	},
+
+	props: {
+		displaySearchBox: {
+			type: Boolean,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			searchText: '',
+			searchResults: [],
+			contactsLoading: false,
+		}
+	},
+
+	computed: {
+		show() {
+			return this.$store.getters.getSidebarStatus()
+		},
+		opened() {
+			return !!this.token && this.show
+		},
+		token() {
+			return this.$route.params.token
+		},
+		conversation() {
+			if (this.$store.getters.conversations[this.token]) {
+				return this.$store.getters.conversations[this.token]
+			}
+			return {
+				token: '',
+				displayName: '',
+				isFavorite: false,
+				type: CONVERSATION.TYPE.PUBLIC,
+				lobbyState: WEBINAR.LOBBY.NONE,
+			}
+		},
+		isSearching() {
+			return this.searchText !== ''
+		},
+		addableUsers() {
+			if (this.searchResults !== []) {
+				const searchResultUsers = this.searchResults.filter(item => item.source === 'users')
+				const participants = this.$store.getters.participantsList(this.token)
+				return searchResultUsers.filter(user => {
+					let addable = true
+					for (const participant of participants) {
+						if (user.id === participant.userId) {
+							addable = false
+							break
+						}
+					}
+					return addable
+				})
+			} return []
+		},
+		addableGroups() {
+			if (this.searchResults !== []) {
+				return this.searchResults.filter((item) => item.source === 'groups')
+			} return []
+		}
+	},
+
+	beforeMount() {
+		this.getParticipants()
+		/**
+		 * If the route changes, the search filter is reset and we get participants again
+		 */
+		EventBus.$on('routeChange', () => {
+			this.searchText = ''
+			this.$nextTick(() => {
+				this.getParticipants()
+			})
+		})
+	},
+
+	methods: {
+		handleClose() {
+			this.$store.dispatch('hideSidebar')
+		},
+		handleInput() {
+			this.contactsLoading = true
+			this.searchResults = []
+			this.debounceFetchSearchResults()
+		},
+
+		debounceFetchSearchResults: debounce(function() {
+			if (this.isSearching) {
+				this.fetchSearchResults()
+			}
+		}, 250),
+
+		async fetchSearchResults() {
+			const response = await searchPossibleConversations(this.searchText)
+			this.searchResults = response.data.ocs.data
+			this.getParticipants()
+			this.contactsLoading = false
+		},
+
+		async toggleGuests() {
```

Comment:
Your lacking lots of comments here.
Keep your functions documented! :rocket:

Reply:
@nickvergessen :sweat_smile: